rust/src/state_machine/phases/sum.rs

```diff
@@ -22,6 +22,13 @@ pub struct Sum {
     sum_dict: SumDict,
 }
 
+#[cfg(test)]
+impl Sum {
```

Comment:
Weirdly that doesn't work because rustc says the fields are private and only accessible in the sum module. I thought within an `impl` they'd be visible wherever that impl is, but it seems that it's not the case.

Reply:
that strange indeed. making them `pub(in module)` should be a feasible workaround.

Reply:
`pub(in crate::state_machine::tests::impls)` ? I've thought about it but it looked kind of ugly/hacky =/

Reply:
it's only needed for tests anyways, i guess some hackyness is ok^^

Reply:
Actually is doesn't work because `crate::state_machine::tests` exists only when compiling the tests. It would need to be `pub(in crate::state_machine)` but I'd prefer to keep the fields private.